src/bin/page_sort.rs

```diff
@@ -0,0 +1,123 @@
+use std::{fs::File, io::{Error, Read, Seek, Write}, path::PathBuf};
+
+use clap::Parser;
+use indicatif::{ProgressBar, ProgressStyle};
+use tracing::{info, warn};
+
+#[derive(Parser, Debug, Clone)]
+struct Arguments {
+    #[arg(short='v', action = clap::ArgAction::Count)]
+    verbose: u8,
+
+    #[arg(long = "no-color", action = clap::ArgAction::SetFalse)]
+    color: bool,
+
+    #[arg(
+        short = 'o',
+        long = "output",
+        default_value = "output",
+        help = "output file name",
+        required = true
+    )]
+    output: PathBuf,
+
+    #[arg(
+        help = "Pages file, should contain one or multiple raw 16K page",
+        value_name = "PAGES_FILE"
+    )]
+    input: PathBuf
+}
+
+fn arr2int(buf:&[u8; 4]) -> u32{
+    ((buf[0] as u32) << 24) | ((buf[1] as u32) << 16) | ((buf[2] as u32) << 8) | (buf[3] as u32)
+}
+
+fn get_page_number(pages: &mut File, offset: u64) -> Result<u64, Error>{
+    let mut buffer = [0; 4];
+    // pages.seek_read(&mut buffer, offset + 4)?;
+    pages.seek(std::io::SeekFrom::Start(offset+4))?;
+    pages.read(&mut buffer)?;
+    Ok(arr2int(&buffer) as u64)
+}
+
+fn copy_page(source: &mut File, destination: &mut File, source_offset: u64, destination_offset: u64) -> Result<(), Error>{
+    let mut buffer = [0; 4096];
+    let mut destination_offset = destination_offset;
+    let mut source_offset = source_offset;
+    loop {
+        // let bytes_read = source.seek_read(&mut buffer, source_offset)?;
+        source.seek(std::io::SeekFrom::Start(source_offset))?;
+        let bytes_read = source.read(&mut buffer)?;
+        if bytes_read == 0 {
+            break;
+        }
+        destination.seek(std::io::SeekFrom::Start(destination_offset))?;
```

Comment:
By the way, seeking beyond EOF is not guaranteed behavior, so we probably should handle this better than just "skipping"